cpp/tests/io/parquet_test.cpp

```diff
@@ -4204,4 +4216,47 @@ TEST_F(ParquetReaderTest, StructByteArray)
   CUDF_TEST_EXPECT_TABLES_EQUAL(expected, result.tbl->view());
 }
 
+TEST_P(ParquetSizedTest, DictionaryTest)
+{
+  constexpr int nrows = 3'000'000;
+  char buf[64];
+
+  auto elements       = cudf::detail::make_counting_transform_iterator(0, [&buf](auto i) {
+    sprintf(buf, "a unique string value suffixed with %d", i % GetParam());
+    return std::string(buf);
+  });
+  auto const col0     = cudf::test::strings_column_wrapper(elements, elements + nrows);
+  auto const expected = table_view{{col0}};
+
+  auto const filepath = temp_env->get_temp_filepath("DictionaryTest.parquet");
+  // set row group size so that there will be only one row group
+  cudf::io::parquet_writer_options out_opts =
+    cudf_io::parquet_writer_options::builder(cudf_io::sink_info{filepath}, expected)
+      .row_group_size_rows(nrows)
+      .row_group_size_bytes(256 * 1024 * 1024);
+  cudf::io::write_parquet(out_opts);
+
+  cudf::io::parquet_reader_options default_in_opts =
+    cudf::io::parquet_reader_options::builder(cudf_io::source_info{filepath});
+  auto const result = cudf_io::read_parquet(default_in_opts);
+
+  CUDF_TEST_EXPECT_TABLES_EQUAL(expected, result.tbl->view());
+
+  // make sure dictionary was used
+  auto const source = cudf::io::datasource::create(filepath);
+  cudf::io::parquet::FileMetaData fmd;
+
+  read_footer(source, &fmd);
+  auto used_dict = [&fmd]() {
+    for (auto enc : fmd.row_groups[0].columns[0].meta_data.encodings) {
+      if (enc == cudf::io::parquet::Encoding::PLAIN_DICTIONARY or
+          enc == cudf::io::parquet::Encoding::RLE_DICTIONARY) {
```

Comment:
Is there a way to see the number of bits used in the dictionary to ensure we aren't using 24 bits for everything?

Reply:
Yeah, I was wondering about that.  For testing I actually printed out the number of bits used :)

The dictionary page header does encode the number of values, so I assume you can infer the number of bits used from that?  Guess I should look at what the reader does.

Reply:
Well that was fun.  In the end it wasn't too hard, but had to go back to Parquet school :wink: